### ISSUE TITLE
VQA finetuning training time

### ISSUE THREAD

**Issue by johntiger1** [airsplay/lxmert#49]
In this section of the readme, it says that fine-tuning should only take 2 hours per epoch:

'''
If no bug came out, then the model is ready to be trained on the whole VQA corpus:

bash run/vqa_finetune.bash 0 vqa_lxr955
It takes around 8 hours (2 hours per epoch * 4 epochs) to converge. The logs and model snapshots will be saved under folder snap/vqa/vqa_lxr955. The validation result after training will be around 69.7% to 70.2%.
'''

Is this with 4 GPUs? Because on my system with a single Titan XP, it is reporting 300+ hours/epoch

![image](https://user-images.githubusercontent.com/14852840/78176958-55325880-742b-11ea-8819-cce2a02067f1.png)

Is this an issue with my system or with the code? Because it seems even with 4 GPUs, we would still need 75 hours/epoch

Thanks

**Comment by airsplay** [comment]
The time is estimated under a single GPU... 

If the original code is used, it seems like a system issue. Some possible reasons: 
- Is the GPU shared with other processes? 
- Does the machine have enough memory? Otherwise, the features would be swapped to disk thus slowdown.

**Comment by johntiger1** [comment]
Ahh, it is my workstation's GPU, so it is unshared. But the workstation only has 16GB memory, while the GPU has 12 GB memory. The MSCOCO image features are larger than 16 GB, but surely we aren't storing the entire dataset in memory all the time?

**Comment by airsplay** [comment]
Actually I stored it in memory for code simplicity. A possible solution might be sequentially loading the data here:
https://github.com/airsplay/lxmert/blob/9b8f0ffd56dba5490f37af9c14f3c112cc10358c/src/tasks/vqa.py#L36
by changing it to `shuffle=False'.

**Comment by johntiger1** [comment]
Oh wow, you had access to a lot more memory than me :P 

I will try that, thanks for the quick reply! 

(I can also send it to a cluster, but I am just trying to do some quick prototyping right now)

**Comment by johntiger1** [comment]
Yes, it seems like training will be very fast on the cluster! 

![image](https://user-images.githubusercontent.com/14852840/78397618-8e033680-75bf-11ea-824c-f2db4d2d1c75.png)


Thanks again

**Comment by airsplay** [comment]
Cool. Great to know that!
